### DISCUSSION TITLE
How to pass negative float as CLI argument?

### DISCUSSION THREAD

**Discussion by wdv4758h** [nushell/nushell#12473] [Q&A]
This works:

```nu
# tmp.nu
def main [value: int] {
  print $value
}
```
```sh
nu tmp.nu -1
```

---

This does not work:

```nu
# tmp.nu
def main [value: float] {
  print $value
}
```
```sh
nu tmp.nu -1
```
```
Error: nu::parser::unknown_flag

  × The `main` command doesn't have flag `-1`.
   ╭─[<commandline>:1:7]
 1 │ main -1
   ·       ┬
   ·       ╰── unknown flag
   ╰────
  help: Available flags: --help(-h). Use `--help` for more information.
```

---

Any way to pass negative float as arguments without hack (like using string and convert inside the program)?

**Comment by ghost** [comment]
you can use `foo (-1.0)` (braces make sure its not interpreted as flag and without `.0` it thinks its an integer

**Reply by wdv4758h** [reply]
I still get error with braces though.

**Reply by ghost** [reply]
i have re-tested this with interactive nu 0.92.1 and 0.91.0 and it worked in both cases for me

```
$ def main [value: float] {
  print $value
}

$ main (-1.0)
-1
```

which nu-version, etc are you using? maybe it broke earlier or later (i cant compile 0.92.2 for now since my rust compiler is outdated)

**Reply by wdv4758h** [reply]
I think you are talking about different use case. What I mean is having a .nu script file, and call that script with arguments. Not defining function in current nushell and call it.

**Reply by ghost** [reply]
ok i can't find a working version for that either - for now either using strings and converting it or flag-arguments (`--value: float` works) seem to be the only option

**Comment by devyn** [comment]
I tested a few things and don't think it's possible. I think maybe you should report it as a bug
